ada_2_fhir-r4/zibs2020/payload/0.0/nl-core-Alert.xsl

```diff
@@ -0,0 +1,181 @@
+2.1 of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+See the GNU Lesser General Public License for more details.
+
+The full text of the license is available at http://www.gnu.org/copyleft/lesser.html
+-->
+
+<xsl:stylesheet exclude-result-prefixes="#all"
+    xmlns="http://hl7.org/fhir"
+    xmlns:util="urn:hl7:utilities" 
+    xmlns:f="http://hl7.org/fhir" 
+    xmlns:xd="http://www.oxygenxml.com/ns/doc/xsl"
+    xmlns:nf="http://www.nictiz.nl/functions" 
+    xmlns:uuid="http://www.uuid.org"
+    xmlns:xs="http://www.w3.org/2001/XMLSchema" 
+    xmlns:xsl="http://www.w3.org/1999/XSL/Transform" 
+    version="2.0">
+    
+    <xsl:output method="xml" indent="yes"/>
+    <xsl:strip-space elements="*"/>
+    
+    <xd:doc scope="stylesheet">
+        <xd:desc>Converts ADA alert to FHIR resource conforming to profile nl-core-Alert</xd:desc>
+    </xd:doc>
+    
+    <xd:doc>
+        <xd:desc>Create a nl-core-Alert instance as a Flag FHIR instance from ADA alert.</xd:desc>
+        <xd:param name="in">ADA element as input. Defaults to self.</xd:param>
+    </xd:doc>
+    <xsl:template match="alert" name="nl-core-Alert" mode="nl-core-Alert" as="element(f:Flag)">
+        <xsl:param name="in" as="element()?" select="."/>
+        <xsl:param name="subject" select="patient/*" as="element()?"/>
+        
+        <xsl:for-each select="$in">
+            <Flag>
+                <xsl:call-template name="insertLogicalId"/>
+                <meta>
+                    <profile value="http://nictiz.nl/fhir/StructureDefinition/nl-core-Alert"/>
+                </meta>
+                
+                <xsl:for-each select="conditie/probleem">
+                    <extension url="http://hl7.org/fhir/StructureDefinition/flag-detail">
+                        <valueReference>
+                            <xsl:call-template name="makeReference">
+                                <xsl:with-param name="in" select="."/>
+                                <xsl:with-param name="profile" select="'nl-core-Problem'"/>
+                            </xsl:call-template>
+                        </valueReference>
+                    </extension>
+                </xsl:for-each>
+                
+                <xsl:for-each select="toelichting">
+                    <extension url="http://nictiz.nl/fhir/StructureDefinition/ext-Comment">
+                        <valueString>
+                            <xsl:call-template name="string-to-string">
+                                <xsl:with-param name="in" select="."/>
+                            </xsl:call-template>
+                        </valueString>
+                    </extension>
+                </xsl:for-each>
+                
+                <status>
```

Comment:
Als ik in een mappings iets met `status` doe typ ik er altijd een toelichting bij vanwege de herleidbaarheid.

Reply:
Done!